scripts/legislation/africa/namibia_english.sh

```diff
@@ -0,0 +1,144 @@
+    sed -E 's/^(PART )V -I/\1VI -/' | \
+    sed -E 's/^(PART VI - ENVIRONMENTAL.*) Objects/\1\n\nObjects/' | \
+    sed -E 's/^(PART )VI -I/\1VII -/' | \
+    sed -E 's/^(PART VII - ENVIRONMENTAL.*) Listing/\1\n\nListing/' | \
+    sed -E 's/^(PART )VII -I/\1VIII -/' | \
+    sed -E 's/^(PART VIII - ENVIRONMENTAL.*) Application/\1\n\nApplication/' | \
+    sed -E 's/^(PART )I -X/\1IX -/' | \
+    sed -E 's/^(PART IX - SPECIAL.*) Consultation/\1\n\nConsultation/' 
+}
+
+function remove_margin_headers {
+    sed -E 's/No. 3966 Government Gazette 27 December 2007//g' | \
+    sed -E 's/Act No. 7, 2007 ENVIRONMENTAL MANAGEMENT ACT, 2007//g' | \
+    sed -E 's/Government Gazette 27 December 2007 No. 3966 //g' 
+}
+
+function identify_leftover_articles {
+  sed -E 's/(in section Article [[]44])/in section 44/' | \
+  sed -E 's/(Article )(\[[0-9]+\])/\n[\2]/g' | \
+  sed -e 's/\[\[\([^]]*\)\]\]/(\1)/g' 
+}
+
+function remove_all_text_after_last_article {
+  sed -E '/^\(58\)/q'
+}
+
+function amend_errors_in_articles {
+  #remove incorrect bullet points
+  sed -E 's/ [[]•]//g' | \
+  #format subsection numbers
+  sed -E 's/(subsection )\(([0-9]+)\)/\1[\2]/g' | \
+  #format (a-z) t0 [a-z]
+  sed -E 's/\(([a-z])\)/[\1]/g' | \
+  #turn (1) to [1] in articles
+  sed -E "s/^(\([0-9]+\\).*) [(]1[)] /\1 [1] /" | \
+  #remove page numbers
+  sed -E 's/. [0-9]+\  /. /g' | \
+  sed -E 's/\.\././g' | \
+
+  #Article 1
+  amend_error_in_article 1 '1995]' '1995])' | \
+  amend_error_in_article 1 '1977]' '1977])' | \
+  amend_error_in_article 1 '2001]' '2001])' | \
+  amend_error_in_article 1 'section 27[(]1[)]' 'section 27[1]' | \
+  #Article 8
+  amend_error_in_article 8 'subsections [(]4[)]' 'subsections [4]' | \
+  #Article 15
+  amend_error_in_article 15 '1991]' '1991])' | \
+  #Article 18
+  amend_error_in_article 18 '\( \[2]' '[2]' | \
+  #Article 19
+  amend_error_in_article 19 'To the extend that this' 'To the extent that this' | \
+  #Article 22
+  amend_error_in_article 22 'person [.]' 'person -' | \
+  #Article 26
+  amend_error_in_article 26 'Enviromental Commissionr' 'Environmental Commissioner' | \
+  #Article 31
+  amend_error_in_article 31 'invalid.PART' 'invalid. PART' | \
+  #Article 38
+  amend_error_in_article 38 '\[37]' '37.' | \
+  #Article 39
+  amend_error_in_article 39 'in section \[44]' 'section 44.' | \
+  amend_error_in_article 39 'initiave' 'initiative' | \
+  #Article 40
+  amend_error_in_article 40 'under section \[42]' 'under section 42.' | \
+  #Article 42
+  amend_error_in_article 42 'section \[44]' 'section 44.' | \
+  #Article 44
+  amend_error_in_article 44 'reasonable time.Appointment' 'reasonable time. Appointment' | \
+  #Article 50
+  amend_error_in_article 50 'otherwise.Appeal' 'otherwise. Appeal' 
+}
+
+function move_article_titles_above_article_bodies {
+  sed -E "s/^(\([0-9]+\\).*)\. /\1.\n\n/" 
+}
+
+function add_newlines_before_headers_and_articles {
```

Comment:
It's helpful to the reader to arrange the functions in a file in the order that they are invoked. For example, this function definition would follow that of `replace_parentheses_around_article_delimiters_with_square_brackets` because that is the order in which the functions are invoked by the `preprocess_state_and_language_input_file` function.

All of that said, why do you need this function? It seems like this is doing something similar to the functions invoked by `apply_common_transformations_to_stdin`. What's different?

Reply:
The difference is there is one less '\n' in this line of the function compared to the one invoked by apply_common_transformations_to_stdin: echo "$stdin" | sed -E ":start;s/${regular_expression}/\1\n\2 \3/;t start". 

Since the article titles were moved down from the article before, I needed 1 newline for the article title to be directly above the article. I couldn't figure out how to remove a newline before a pattern so I simply modified the add_newlines_before_headers_and_articles function to enter 1 newline instead of 2.

Reply:
I see. In that case, does this function need to invoke `get_line_prefix_regular_expression` or would `get_article_regular_expression` suffice?

Reply:
I'm not too sure what you mean by this. But replacing `get_line_prefix_regular_expression` with `get_article_regular_expression` in the function returns an error.